.travis.yml

```diff
@@ -1,57 +1,221 @@
 language: cpp
 
-install:
-  - pip install --user conan cmake
-
 jobs:
   include:
-    - os: osx
+    - os: osx       # 10.15.4 - gcc-9
+      compiler: gcc
+      osx_image: xcode11.4
+      env:
+        - MACOSX_DEPLOYMENT_TARGET=10.15
+        - GCC_VER="9"
+        - MATRIX_EVAL="CC=gcc-${GCC_VER} && CXX=g++-${GCC_VER}"
+
+    - os: osx       # 10.14.6 - gcc-9
       compiler: gcc
-      osx_image: xcode11.2    # includes gcc-9 by default
+      osx_image: xcode11.3
       env:
-        # Conan is at: ${HOME}/Library/Python/2.7/bin: we need this in the path
-        - PATH="${HOME}/Library/Python/2.7/bin:${PATH}"
+        - MACOSX_DEPLOYMENT_TARGET=10.14
         - GCC_VER="9"
         - MATRIX_EVAL="CC=gcc-${GCC_VER} && CXX=g++-${GCC_VER}"
+
+    - os: osx       # 10.13 - gcc-9
+      compiler: gcc
+      osx_image: xcode10.1
+      env:
+        - MACOSX_DEPLOYMENT_TARGET=10.13
+        - GCC_VER="9"
+        - MATRIX_EVAL="CC=gcc-${GCC_VER} && CXX=g++-${GCC_VER}" #brew install gcc${GCC_VER} &&
+
+    - os: osx       # 10.15.4 - clang
+      compiler: clang
+      osx_image: xcode11.4
+      env:
+        - MACOSX_DEPLOYMENT_TARGET=10.15
+
+    - os: osx       # 10.14.6 - clang
+      compiler: clang
+      osx_image: xcode11.3
+      env:
+        - MACOSX_DEPLOYMENT_TARGET=10.14
+
+    - os: osx       # 10.13 - clang
+      compiler: clang
+      osx_image: xcode10.1
+      env:
+        - MACOSX_DEPLOYMENT_TARGET=10.13
+
+    - os: osx       # 10.15.4 - clang sanitize-all-the-things build
+      compiler: clang
+      osx_image: xcode11.4
+      env:
+        - MACOSX_DEPLOYMENT_TARGET=10.15
+        - CONFIG_OPTS="-D ENABLE_CPPCHECK:BOOL=ON \
+                       -D ENABLE_SANITIZER_ADDRESS:BOOL=ON \
+                       -D ENABLE_SANITIZER_MEMORY:BOOL=ON \
+                       -D ENABLE_SANITIZER_THREAD:BOOL=ON \
+                       -D ENABLE_SANITIZER_UNDEFINED_BEHAVIOR:BOOL=ON"
```

Comment:
This kind of test should probably be an EXPECT_FAIL kind of thing. I think we don't really want to run more than one thread sanitizer at once.
We should run all the sanitizers, but in sequence, not simultaneously.

Reply:
Yeah, definitely have to do something about the sanitizers, they are not compatible together. The build will either warn or fail here, because the compiler doesn't allow it (but CMakeLists is supposed to check for it)

Reply:
This particular build did not fail: https://travis-ci.org/github/lefticus/cpp_starter_project/builds/690480254

I think #26 is meant to address this issue?